pkg/utils/baremetal/bootstrap.go

```diff
@@ -106,5 +108,5 @@ func WaitForBaremetalBootstrapControlPlane(ctx context.Context, config *rest.Con
 		return fmt.Errorf("failed to persist masters file to disk: %w", err)
 	}
 
-	return withSyncErr
+	return errors.Wrap(withSyncErr, asset.ControlPlaneCreationError)
```

Comment:
```suggestion
	return fmt.Errorf("%s: %w", asset.ControlPlaneCreationError, withSyncErr)
```

Reply:
d'oh how embarrassing, as I know better. The problem is we need to return `nil` if `withSyncErr` is `nil`, so I think we need to go a bit further than the suggestion?

Reply:
Ah yes.
```suggestion
    if withSyncErr != nil {
	    return fmt.Errorf("%s: %w", asset.ControlPlaneCreationError, withSyncErr)
	}
	return nil
```

Reply:
should be fixed now